File: entries/sha256-file/go/main.go
package main

import (
	"crypto/sha256"
	"fmt"
	"io"
	"os"
)

func main() {
	f, err := os.Open(os.Args[1])
	if err != nil {
		os.Exit(1)
	}
	h := sha256.New()
	if _, err := io.Copy(h, f); err != nil {
		os.Exit(1)
	}
	fmt.Printf("%x\n", h.Sum(nil))
}

File: entries/sha256-file/python/main.py
#!/usr/bin/env python3
"""Print the SHA-256 digest of a file (see tasks/sha256-file/spec.md)."""
import hashlib
import sys


def main():
    try:
        with open(sys.argv[1], "rb") as f:
            digest = hashlib.file_digest(f, "sha256").hexdigest()
    except OSError:
        sys.exit(1)
    print(digest)


if __name__ == "__main__":
    main()
